runtime/src/main/java/org/corfudb/runtime/object/transactions/WriteSetSMRStream.java

```diff
@@ -167,10 +167,17 @@ void mergeTransaction() {
         if (Address.nonAddress(currentContextPos)) {
             currentContextPos = -1;
         }
-        return Collections.singletonList(contexts
-                .get(currentContext)
-                .getWriteSetEntryList(id)
-                .get((int)(currentContextPos)));
+
+        try {
+            return Collections.singletonList(contexts
+                    .get(currentContext)
+                    .getWriteSetEntryList(id)
```

Comment:
if I were to tighten this up, I would just have getWriteSetEntry(id, currentContextPost) or something to that effect. This leaves a lose end for other bugs.